### ISSUE TITLE
corrupte template definitions after upgrading to 7.2.0

### ISSUE THREAD

**Issue by SCullman** [2sic/2sxc#289]
After upgrading to 7.2.0, several apps are failing. 

Error: App is currently unavailable. DotNetNuke.Services.Exceptions.ModuleLoadException: Object reference not set to an instance of an object. ---> System.NullReferenceException: Object reference not set to an instance of an object. at ToSic.SexyContent.Template.get_UseForList() at ToSic.SexyContent.SexyControlEditBase.get_IsList() at ToSic.SexyContent.SexyViewContentOrApp.get_ModuleActions() --- End of inner exception stack trace ---

It looks like the definition of the template views are broken. Content type and template path are not listed anymore.
![brokentemplate](https://cloud.githubusercontent.com/assets/5693190/8673465/fa05851a-2a36-11e5-827e-22b82960943d.JPG)

**Comment by iJungleboy** [comment]
It may be an upgrade issue (incomplete upgrade) - we've had a few and are fixing them in 7.2.2. Should be released this week, please check that.

**Comment by iJungleboy** [comment]
We figured it out. Apparently DNN causes multiple upgrades to run in parallel - which caused side effects. We're now trying to build locks into the system to prevent this, not quite done yet. Probably Monday.

**Comment by raphael-m** [comment]
While upgrading to v7 from earlier versions, the upgrade accidentally (soft-)deletes some templates (the cause is actually the 'overlapping recycle' in IIS which causes DNN to run upgrades multiple times at the same time). We just released 2sxc 7.2.2 which fixes the ugprade. We recommend to start the upgrade again from version 6 by restoring a backup.

If you can't restore the backup, you can also try to recover the templates by running the following SQL:
```SQL
-- Undelete all values accidentally deleted by the upgrade code for 07.00.00
UPDATE ToSIC_EAV_Values
SET ChangeLogDeleted = NULL
FROM ToSIC_EAV_Values INNER JOIN
    ToSIC_EAV_ChangeLog ON ToSIC_EAV_Values.ChangeLogDeleted = ToSIC_EAV_ChangeLog.ChangeID
WHERE (ToSIC_EAV_ChangeLog.[User] = N'System-ModuleUpgrade-070000') AND
((SELECT COUNT(*) AS ExistingValuesCount
FROM ToSIC_EAV_Values AS v2
WHERE (ValueID = ToSIC_EAV_Values.ValueID) AND (EntityID = ToSIC_EAV_Values.EntityID) AND (ChangeLogDeleted IS NULL)) = 0)
```
